src/Platforms/OMPTarget/ompBLAS.cpp

```diff
@@ -23,6 +23,141 @@ namespace qmcplusplus
 namespace ompBLAS
 {
 
+template<typename T>
+ompBLAS_status gemm_impl(ompBLAS_handle& handle,
+                         const char transa,
+                         const char transb,
+                         const int M,
+                         const int N,
+                         const int K,
+                         const T alpha,
+                         const T* const A,
+                         const int lda,
+                         const T* const B,
+                         const int ldb,
+                         const T beta,
+                         T* const C,
+                         const int ldc)
+{
+  if (transa == 'T' && transb == 'N') //A(ji) * B(jk) -> C(ik)
+  {
+    PRAGMA_OFFLOAD("omp target teams distribute collapse(2) num_teams(M * N) is_device_ptr(A, B, C)")
+    for (size_t m = 0; m < M; m++)
+      for (size_t n = 0; n < N; n++)
+      {
+        C[n * ldc + m] *= beta;
```

Comment:
When beta is `0` the result should be `0`. However in practice, if `C[n * ldc + m]` is `NaN`, the result is still `NaN`.  So change to
```
C[n * ldc + m] = beta == T(0) ? T(0) : C[n * ldc + m] * beta;
```

Reply:
ok, I fixed this (I suppose if beta==0, we shouldn't care about whether C is initialized to something sensible anyway)